src/sys/time.rs

```diff
@@ -181,6 +182,191 @@ impl PartialEq for TimeVal {
 
 impl Eq for TimeVal { }
 
+#[derive(Clone, Copy)]
+pub struct TimeSpec(pub timespec);
+
+impl AsRef<timespec> for TimeSpec {
+    fn as_ref(&self) -> &timespec {
+        &self.0
+    }
+}
+
+impl TimeSpec {
+    #[inline]
+    pub fn zero() -> TimeSpec {
```

Comment:
I'd prefer to hold off adding all these methods until #277 is decided, which would likely mostly be interconversion with [`std::time::Duration`](http://doc.rust-lang.org/std/time/struct.Duration.html).